research-software-directory/RSD-as-a-service
# Scraper improvements

# Improve GIT scrapers

Changes proposed in this pull request:

* Each GIT scraper now only pushes (via its own PATCH request per repo to scrape) its own data, versus pushing all data (via a single POST request)
* The repos are, since the data is now independent, scraped concurrently
* Add a dedicated class that aggregates commits per week, commit scrapers are now expected to return an instance of that class instead of a (JSON) String

How to test:
* `docker-compose down --volumes && docker-compose build --parallel && docker-compose up --scale data-generation=1`
* Wait or run the GIT scrapers manually a few times: 
	* `docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.git.MainProgrammingLanguages`
	* `docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.git.MainLicenses`
	* `docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.git.MainCommits`
* Check that the scrapers run without errors
* Check that the commit graphs and language bars are correct

PR Checklist:
